packages/gatsby-cli/src/reporter/index.ts

```diff
@@ -40,76 +58,103 @@ if (process.env.GATSBY_LOGGER.includes(`json`)) {
   require(`./loggers/ink`)
 }
 
-const util = require(`util`)
-const { stripIndent } = require(`common-tags`)
-const chalk = require(`chalk`)
-const { trackError } = require(`gatsby-telemetry`)
-const tracer = require(`opentracing`).globalTracer()
-
-const { getErrorFormatter } = require(`./errors`)
-const { getStore } = require(`./redux`)
-import constructError from "../structured-errors/construct-error"
-
 const errorFormatter = getErrorFormatter()
 
-import type { ActivityTracker, ActivityArgs, Reporter } from "./types"
+// eslint-disable-next-line @typescript-eslint/interface-name-prefix
+export interface ActivityTracker {
+  start(): void
+  end(): void
+  span: Span
+  setStatus(status: string): void
+  panic: (errorMeta: ErrorMeta, error?: Error) => never
+  panicOnBuild: (errorMeta: ErrorMeta, error?: Error) => Error | undefined
+}
 
-const addMessage = level => text => reporterActions.createLog({ level, text })
+export type ProgressActivityTracker = Omit<ActivityTracker, "end"> & {
+  tick(increment?: number): void
+  done(): void
+  total: number
+}
 
-let isVerbose = false
+export interface IActivityArgs {
+  parentSpan?: Span
+  id?: string
+}
 
-const interruptActivities = () => {
-  const { activities } = getStore().getState().logs
-  Object.keys(activities).forEach(activityId => {
-    const activity = activities[activityId]
-    if (
-      activity.status === ActivityStatuses.InProgress ||
-      activity.status === ActivityStatuses.NotStarted
-    ) {
-      reporter.completeActivity(activityId, ActivityStatuses.Interrupted)
-    }
-  })
+export interface IErrorDetails {
+  id?: ErrorId
+  context?: Record<string, unknown>
+  error?: Error
+  [key: string]: unknown
 }
 
-const prematureEnd = () => {
-  // hack so at least one activity is surely failed, so
-  // we are guaranteed to generate FAILED status
-  // if none of activity did explicitly fail
-  reporterActions.createPendingActivity({
-    id: `panic`,
-    status: ActivityStatuses.Failed,
-  })
+export type ErrorMeta = string | IErrorDetails | IErrorDetails[] | Error
 
-  interruptActivities()
+interface ILogAction {
+  type: Actions.Log
+  payload: Record<string, unknown>
 }
 
-signalExit((code, signal) => {
-  if (code !== 0 && signal !== `SIGINT` && signal !== `SIGTERM`) prematureEnd()
-  else interruptActivities()
-})
```

Comment:
it looks like this was lost in the translation?

Reply:
including the interruptActivities and prematureEnd stuff. 

Why are we getting rid of it all?

Reply:
Oh I see you added these to the class. I don't think we want to expose those APIs. 

Can we move them out of the class and put them top level again?

Reply:
They're in the class because they refer to the instance, not because they're exported. I'm also trying to minimise the top level side effects.